oxidauth-rs/src/client/auth/oauth2/redirect.rs

```diff
@@ -0,0 +1,31 @@
+pub use oxidauth_http::{
+    response::Response,
+    server::api::v1::auth::{
+        oauth2::redirect::Oauth2RedirectRes,
+        register::{AuthorityStrategy, RegisterReq, RegisterRes},
+    },
+};
+use oxidauth_kernel::{auth::oauth2::redirect::Oauth2RedirectParams, error::BoxedError};
+
+pub use oxidauth_usecases::auth::strategies::*;
+
+use super::*;
+
+impl Client {
+    #[tracing::instrument(skip(self))]
+    pub async fn oauth2_redirect<T>(
+        &self,
+        params: T,
+    ) -> Result<Response<Oauth2RedirectRes>, BoxedError>
+    where
+        T: Into<Oauth2RedirectParams> + fmt::Debug,
+    {
+        let params = params.into();
+
+        let result: Response<Oauth2RedirectRes> = self
+            .post("/auth/oauth2/redirect", params)
+            .await?;
+
+        Ok(result)
+    }
+}
```

Comment:
Not totally sure I understand what this is for..? Fetching the redirect url, yeah?

Reply:
yes. Navi is getting the redirect url through oxidauth-js so I don't technically need this, but I thought it might be good to provide this as an option? Can remove no problem tho if not needed/desired

Reply:
Hmm... Let me think about this. I know we discussed it a little before but I'm wondering about any security issues with keys being leaked here.

Reply:
Though -- I suppose it's all publicly accessible anyways...

Reply:
I could also see how if we don't obviously need it, we shouldn't add it yet. Totally fine to remove